packages/kbn-management/settings/components/field_row/footer/reset_link.tsx

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+import { EuiLink } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+import type {
+  FieldDefinition,
+  SettingType,
+  UnsavedFieldChange,
+} from '@kbn/management-settings-types';
+import { isFieldDefaultValue } from '@kbn/management-settings-utilities';
+
+/**
+ * Props for a {@link InputResetLink} component.
+ */
+export interface InputResetLinkProps<T extends SettingType> {
+  /** The {@link FieldDefinition} corresponding the setting. */
+  field: Pick<
+    FieldDefinition<T>,
+    'ariaAttributes' | 'id' | 'savedValue' | 'isOverridden' | 'defaultValue' | 'type'
+  >;
+  /** A handler for when a field is reset to its default or saved value. */
+  onReset: () => void;
+  /** A change to the current field, if any. */
+  unsavedChange?: UnsavedFieldChange<T>;
+}
+
+export const DATA_TEST_SUBJ_RESET_PREFIX = 'management-settings-resetField';
+/**
+ * Component for rendering a link to reset a {@link FieldDefinition} to its default
+ * or saved value.
+ */
+export const InputResetLink = <T extends SettingType>({
+  onReset: onClick,
+  field,
+  unsavedChange,
+}: InputResetLinkProps<T>) => {
+  if (isFieldDefaultValue(field, unsavedChange) || field.isOverridden) {
```

Comment:
According to this check, we would display the "Reset to default" link, even if the saved value is equal to the default value:

<img width="1261" alt="Screenshot 2023-09-19 at 12 22 12" src="https://github.com/elastic/kibana/assets/59341489/1896e47f-058b-4ab4-a91e-cb91735ebd2c">

Overall, I think this is fine, but just wanted to note that I think in the existing Advanced settings page, the link is only displayed if the saved value is different from the default value (i.e. the unsaved value is not considered):

<img width="1176" alt="Screenshot 2023-09-19 at 12 22 57" src="https://github.com/elastic/kibana/assets/59341489/56d6523d-8121-4762-9248-49cac44bd190">

So I was wondering if we should change the logic of `isFieldDefaultValue` to only compare the saved value with the default value.

Reply:
I guess it depends on if resetting the field refers to unsaved changes, too.  I'm AFK, but can check the behavior of the current page.

Reply:
I believe in the original Advanced settings app, the link is only displayed if the saved value is different from the default value - it doesn't consider the unsaved value. So if we just make some change, the link is not displayed (only the yellow dot indicates that there is some change).

I guess this is more of a design choice of whether we want to keep it as it is in Advanced settings now, or whether we also want to display the link for unsaved changes.

Reply:
Apologies if my explanation wasn't clear, I hope these recordings below show what I meant. In the first video, the link doesn't show up until we save the change, while in the second it shows up immediately after the change.

https://github.com/elastic/kibana/assets/59341489/16057223-415a-46bd-9f68-2f38c2537f08



https://github.com/elastic/kibana/assets/59341489/70fac7c3-afe3-4337-b512-14581a7c565a

Reply:
Actually, after thinking about it, it makes sense to keep the current behaviour in this PR, so that if the user makes several changes and, before saving them, decides to revert one of them to its default value, they would be able to do so by clicking on the reset link.